lichess-bot.py

```diff
@@ -214,12 +219,20 @@ def start(li, user_profile, config, logging_level, log_filename, one_game=False)
                     logger.info(f"Accept {chlng}")
                     queued_processes += 1
                     li.accept_challenge(chlng.id)
+                    if challenge_id == chlng.id:
+                        challenge_id = None
                     logger.info(f"--- Process Queue. Total Queued: {queued_processes}. Total Used: {busy_processes}")
                 except (HTTPError, ReadTimeout) as exception:
                     if isinstance(exception, HTTPError) and exception.response.status_code == 404:  # ignore missing challenge
                         logger.info(f"Skip missing {chlng}")
                     queued_processes -= 1
 
+            if queued_processes + busy_processes < max_games and not challenge_queue and (config.get("matchmaking") or {}).get("allow_matchmaking") and ((last_challenge_created + ((config.get("matchmaking") or {}).get("challenge_interval") or 30) * 60) < time.time() or last_challenge_created + challenge_expire_time < time.time() and challenge_id):
```

Comment:
This conditional is too long. Everything after `and not challenge_queue` should be placed in a `Matchmaker` method named `matchmaker.should_create_challenge()` or similar.

The new variables `last_challenge_created`, `challenge_expire_time`, and `challenge_id` should be members of the `Matchmaker` class and initialized in that class' `__init__()`, as well.

After those changes,
```python
 if queued_processes + busy_processes < max_games and not challenge_queue and matchmaker.should_create_challenge():
    logger.debug("Challenging a random bot")
    matchmaker.challenge()
```